notebooks/network-analysis.ipynb

```diff
@@ -0,0 +1,470 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create reverse connection dictionary for each subnetwork\n",
+    "# tailwater-segment-id: {segment-id: [upstream-connected-segment-id]}\n",
+    "subnets_ = nhd_network.reachable_network(rconn)\n",
+    "\n",
+    "# create a dictionary, `subreaches` containing lists of all linear connected segments between junctions (reaches) in the supernetwork.\n",
+    "# Recall, a reach is composed of several interconnected segments between junctions.\n",
+    "# tailwater-segment-id: [[rch1-segment-id-1, rch1-segment-id-2, ..., rch1-segment-id-n],[rch2-segment-id1, rch2-segment-id2, ..., rch2-segment-id-n]]\n",
+    "subreaches = {}\n",
+    "for tw, net in subnets_.items():  # for the number of tailwaters in the supernetwork\n",
+    "\n",
+    "    # I dont know what this does... something with identifying junctions\n",
+    "    path_func = partial(nhd_network.split_at_junction, net)\n",
+    "\n",
+    "    # conduct a depth-first search to decompose the network into list of reaches\n",
+    "    reach = nhd_network.dfs_decomposition(nhd_network.reverse_network(net), path_func)\n",
+    "\n",
+    "    # add reach list for subnetwork tw to the subreaches library\n",
+    "    subreaches[tw] = reach\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualize reaches in a single subnetwork\n",
+    "- Make a map showing each of the reaches in the Colorado River Basin subnetwork"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# extract a single large subnetwork from the supernetwork GeoDataFrame.\n",
+    "# Turns out that subnetwork code 4 is the Colorado River Basin\n",
+    "# !!! TODO: Create some sort of look up table to query river basins by name, rather than some arbitrary number\n",
+    "sub_gdf = gdf[gdf[\"subnetwork\"] == 4]\n",
+    "\n",
+    "# get reaches in the subnetwork\n",
+    "rchs = subreaches[list(sub_gdf[[\"to\"]].idxmin())[0]]\n",
+    "\n",
+    "# create a reach-code column in the GeoDataFrame\n",
+    "sub_gdf[\"reach\"] = 0\n",
+    "\n",
+    "# loop through all reaches in the subnetwork, populate the newly created \"reach\" column with an integer reach identification code\n",
+    "i = 1\n",
+    "for r in rchs:\n",
+    "\n",
+    "    # get the stream segments in subnetwork `sub`\n",
+    "    A = [abs(ele) for ele in list(r)]\n",
```

Comment:
This can just be `list(map(abs, r))` or `np.abs(r)`
In fact, we already know that the negative values won't be in the index, so we could just drop them?